src/anomalib/models/components/base/export_mixin.py

```diff
@@ -235,20 +239,57 @@ def to_openvino(
             ov_args = {} if ov_args is None else ov_args
 
             model = ov.convert_model(model_path, **ov_args)
+            model_input = model.input(0)
+
             if compression_type == CompressionType.INT8:
                 model = nncf.compress_weights(model)
             elif compression_type == CompressionType.INT8_PTQ:
                 if datamodule is None:
                     msg = "Datamodule must be provided for OpenVINO INT8_PTQ compression"
                     raise ValueError(msg)
 
-                dataloader = datamodule.val_dataloader()
+                if model_input.partial_shape[0].is_static:
+                    datamodule.train_batch_size = model_input.shape[0]
+                dataloader = datamodule.train_dataloader()
                 if len(dataloader.dataset) < 300:
                     logger.warning(
                         f">300 images recommended for INT8 quantization, found only {len(dataloader.dataset)} images",
                     )
+
                 calibration_dataset = nncf.Dataset(dataloader, lambda x: x["image"])
                 model = nncf.quantize(model, calibration_dataset)
+            elif compression_type == CompressionType.INT8_ACQ:
```

Comment:
Can we extract nncf related code to another method? Right now it bloats the ov export method.